front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts

```diff
@@ -115,9 +115,9 @@ export abstract class TransactionTypeBaseComponent implements OnInit, OnDestroy
         switchMap(([contribution_date, contactId]) => {
           const aggregation_group: AggregationGroups | undefined = (transactionType?.transaction as SchATransaction)
```

Comment:
what do you think of moving this aggregation_group assignment into `getPreviousTransaction` since it's derived from `transactionType` which you are now passing in.  We'd have to update service function to not make a network call if it's undefined, but maybe that's where the condition should be anyways right?  let the component ask the service "i'd like the previous transaction" and have the service decide, based on the args, how much work it should do to return a value.  It would make this observable flow easier to read for sure

Reply:
Good suggestion. This refactoring has been made for this PR.